src/components/CookieConsentBanner.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { Box, Snackbar as SnackbarUI, Typography, useTheme } from '@mui/material';
+
+import TextWithLink from 'src/components/common/TextWithLink';
+import Button from 'src/components/common/button/Button';
+import { useDocLinks } from 'src/docLinks';
+import { useLocalization, useUser } from 'src/providers';
+import { requestUserCookieConsentUpdate } from 'src/redux/features/user/usersAsyncThunks';
+import { useAppDispatch } from 'src/redux/store';
+import strings from 'src/strings';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+
+const ONE_YEAR_IN_MILLISECONDS = 365 * 24 * 60 * 60 * 1000;
+
+export default function CookieConsentBanner() {
+  const { activeLocale } = useLocalization();
+  const { isMobile } = useDeviceInfo();
+  const { user } = useUser();
+  const theme = useTheme();
+  const dispatch = useAppDispatch();
+  const docLinks = useDocLinks();
+
+  const [confirmed, setConfirmed] = useState(false);
+  const [visible, setVisible] = useState(false);
+
+  const updateUserCookieConsent = useCallback((consent: boolean) => {
+    dispatch(requestUserCookieConsentUpdate({ cookiesConsented: consent }));
+    setConfirmed(true);
+    setVisible(false);
+  }, []);
+
+  useEffect(() => {
+    const cookiesConsentedTimeIsGreaterThanOneYear = user?.cookiesConsentedTime
+      ? Date.now() - new Date(user.cookiesConsentedTime).valueOf() > ONE_YEAR_IN_MILLISECONDS
+      : false;
+
+    if (user && (typeof user?.cookiesConsented !== 'boolean' || cookiesConsentedTimeIsGreaterThanOneYear)) {
+      setVisible(true);
+    }
+  }, [user]);
```

Comment:
It seems like you can return out early if there's no `user.cookiesConsentedTime`

Something like this might be easier to read:
```
useEffect(() => {
    if (!(user && user.cookiesConsentedTime)) {
      return;
    }

    const cookiesConsentedTimeIsGreaterThanOneYear =
      Date.now() - new Date(user.cookiesConsentedTime).valueOf() > ONE_YEAR_IN_MILLISECONDS;

    if (cookiesConsentedTimeIsGreaterThanOneYear) {
      setVisible(true);
    }
  }, [user]);
```

Reply:
This won't work as-is because we also need to display the cookie consent banner when `user.cookiesConsented` is `null` (before the user has made a selection). I'll see what I can do to improve how this reads using your suggestion.

Reply:
We also need to ask the user again after one year has passed.

Reply:
> This won't work as-is because we also need to display the cookie consent banner when `user.cookiesConsented` is `null` (before the user has made a selection). I'll see what I can do to improve how this reads using your suggestion.

I see, it sounds like that functionality should be another use effect.

Reply:
> if (cookiesConsentedTimeIsGreaterThanOneYear) {
>       setVisible(true);
>     }

Is that not handled by this part?
```
    if (cookiesConsentedTimeIsGreaterThanOneYear) {
      setVisible(true);
    }
```

Reply:
@nickgraz I'm not sure I see why or agree that it needs to be another `useEffect`. It seems perfectly reasonable to handle these two cases in one `useEffect` in my mind since both cases are related to the user's cookie consent.

Reply:
I modified the `useEffect` based on your suggestion to improve readability a bit, but this `useEffect` is working now and the code doesn't seem particularly complex or clever so I think this can stay as-is.

Reply:
Looks good to me. I generally try to split up business logic into separate effects, because then you can have more targeted effects, which means less re-execution due to changed dependencies. "Show the banner if the user hasn't told us yes or no" and "show the banner if the consent is too old" are separate requirements that are looking at different pieces of data. 

Either way, this is good the way you have it now, I am probably just over thinking it!